clang/lib/3C/CheckedRegions.cpp

```diff
@@ -154,13 +159,26 @@ bool CheckedRegionFinder::VisitDoStmt(DoStmt *S) {
 
 bool CheckedRegionFinder::VisitCompoundStmt(CompoundStmt *S) {
   // Visit all subblocks, find all unchecked types
-  bool Localwild = 0;
+  bool Localwild = false;
   for (const auto &SubStmt : S->children()) {
     CheckedRegionFinder Sub(Context, Writer, Info, Seen, Map, EmitWarnings);
     Sub.TraverseStmt(SubStmt);
     Localwild |= Sub.Wild;
   }
 
+  // If we are a function def, need to check return type
+  if (isTopLevel(Context, S)) {
+    const auto &Parents = Context->getParents(*S);
+    assert(!Parents.empty());
+    FunctionDecl* Parent = const_cast<FunctionDecl*>(Parents[0].get<FunctionDecl>());
```

Comment:
Why does `Parent` need to be non-`const` ?

Reply:
Because `ProgramInfo::getVariable` takes a non-`const`

Reply:
I think we might be able to make the parameter of `getVariable` `const`. We should rarely if ever be mutating this sort of clang AST node object.  Perhaps not as part of this PR, but in a later change we could add `const` annotation to these AST node objects where we can.

Reply:
Ok, I'm happy to go through and try to make that change.

Reply:
But, per the status meeting: Let's just do this as a separate PR; leave the `const_cast` for now.